dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ranger/RangerConnector.java

```diff
@@ -0,0 +1,284 @@
+import com.google.edwmigration.dumper.application.dumper.handle.AbstractHandle;
+import com.google.edwmigration.dumper.application.dumper.handle.Handle;
+import com.google.edwmigration.dumper.application.dumper.task.AbstractTask;
+import com.google.edwmigration.dumper.application.dumper.task.Task;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import com.google.edwmigration.dumper.application.dumper.utils.ArchiveNameUtil;
+import com.google.edwmigration.dumper.plugin.ext.jdk.annotation.Description;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.Group;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.GroupsFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.PoliciesFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.Policy;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.Role;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.RolesFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.Service;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.ServicesFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.User;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RangerDumpFormat.UsersFormat;
+import com.google.errorprone.annotations.ForOverride;
+import java.io.Writer;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.util.Iterator;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RespectsInput(
+    order = 100,
+    arg = ConnectorArguments.OPT_HOST,
+    description = "The hostname of the Ranger server.",
+    defaultValue = ConnectorArguments.OPT_HOST_DEFAULT)
+@RespectsInput(
+    order = 101,
+    arg = ConnectorArguments.OPT_PORT,
+    description = "The port of the Ranger server.",
+    defaultValue = ConnectorArguments.OPT_RANGER_PORT_DEFAULT)
+@AutoService({Connector.class})
+@Description("Dumps services and policies from Apache Ranger.")
+public class RangerConnector extends AbstractConnector {
+
+  @SuppressWarnings("UnusedVariable")
+  private static final Logger LOG = LoggerFactory.getLogger(RangerConnector.class);
+
+  public RangerConnector() {
+    super("ranger");
+  }
+
+  @Nonnull
+  @Override
+  public String getDefaultFileName(boolean isAssessment, Clock clock) {
+    return ArchiveNameUtil.getFileName(getName());
+  }
+
+  @Override
+  public void addTasksTo(List<? super Task<?>> out, @Nonnull ConnectorArguments arguments) {
+    out.add(new DumpGroupsTask());
+    out.add(new DumpPoliciesTask());
+    out.add(new DumpRolesTask());
+    out.add(new DumpServicesTask());
+    out.add(new DumpUsersTask());
```

Comment:
I would suggest adding standard dumper metadata tasks - `DumpMetadataTask` and `FormatTask`. Possibly in a separate PR.

https://github.com/google/dwh-migration-tools/blob/42e48dc2c8d6e2e1220373ace09018059f88a9c8/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/HadoopMetadataConnector.java#L67-L68

Reply:
Good point, I'll add these in a subsequent PR.